Clean documentation (#2065)
kyma-project/api-gateway: docs/user/README.md

The API Gateway module offers the following features:

- API Exposure: The module uses Istio features to help you easily and securely expose your workloads by creating APIRule custom resources. With an APIRule, you can:
  - Group multiple workloads and expose them under a single host.
  - Use a short host name to simplify the migration of resources to a new cluster.
  - Configure the **noAuth** access strategy, which offers a simple configuration to allow access to specific HTTP methods.
  - Secure your workloads by configuring **jwt** or **extAuth** access strategies. The **jwt** access strategy enables you to use Istio's JWT configuration to protect your exposed services and interact with them using JSON Web Tokens. The **extAuth** access strategy allows you to implement custom authentication and authorization logic.

- Gateway configuration:
  - Default Kyma Gateway: The module sets up the default TLS Kyma Gateway, which uses the default domain and a self-signed certificate.

>   - Custom Gateway: The module allows you to configure a custom Gateway. For production environments, it is recommended to use a custom Gateway. Additionally, it enables you to expose workloads using a custom domain and DNSEntry. 

Reviewer: ```suggestion
  - Custom Gateway: The module allows you to configure a custom Gateway, which is recommended for production environments. Additionally, it enables you to expose workloads using a custom domain and DNSEntry. 
```

The API Gateway module offers the following features:

> - API Exposure: Easily and securely expose your workloads by creating APIRule custom resources, which leverage Istio capabilities through the APIRule CustomResourceDefinition.

Reviewer: ```suggestion
- API Exposure: The module uses Istio features to help you easily and securely expose your workloads by creating APIRule custom resources. With an APIRule, you can:
  - Group multiple workloads and expose them under a single host.
  - Use a short host name to simplify the migration of resources to a new cluster.
  - Configure the **noAuth** access strategy, which offers a simple configuration to allow access to specific HTTP methods.
  - Secure your workloads by configuring **jwt** or **extAuth** access strategies. The **jwt** access strategy enables you to use Istio's JWT configuration to protect your exposed services and interact with them using JSON Web Tokens. The **extAuth** access strategy allows you to implement custom authentication and authorization logic.
```

### APIGateway Controller

APIGateway Controller handles the configuration of Kyma Gateway and the resources defined in the APIGateway custom resource (CR). The controller is responsible for:
- Configuring Kyma Gateway
- Managing Certificate and DNSEntry resources

### APIRule Controller

> APIRule Controller uses [Istio](https://istio.io/) resource to expose and secure APIs.

Reviewer: ```suggestion
APIRule Controller uses [Istio](https://istio.io/) resources to expose and secure APIs.
```
I guess it uses multiple resources, right? like AuthorizationPolicy, VirtualService, and RequestAuthentication
